### DISCUSSION TITLE
Is that possible to relation to different model with a field by some other field value?

### DISCUSSION THREAD

**Discussion by Riant** [prisma/prisma#18941] [Q&A] [answered]
Is that possible  to relation to different model with a field by some other field value like below:

```
model Log {
  id         Int          @id @default(autoincrement()) @db.UnsignedInt
  type       String   @db.VarChar(16)
  link_id    Int?      @db.UnsignedInt
  A          A?         @relation("aLog", fields: [type="a", link_id])  // here
  B          B?         @relation("bLog", fields: [type="b", link_id])  // here
}

model A {
  id         Int        @id @default(autoincrement()) @db.UnsignedInt
  Logs     Log[]    @relation("aLog")
}

model B {
  id         Int       @id @default(autoincrement()) @db.UnsignedInt
  Logs     Log[]  @relation("bLog")
}
```

Thanks.

**Comment by nurul3101** [comment]
Hello @Riant 👋 

Can you please elaborate on what you are trying to achieve? I didn't understand your question.

**Reply by nurul3101** [reply]
I am checking to see if you had a chance to look at my last response. Please let us know in case you need any assistance.

**Reply by Riant** [reply]
Hi @nurul3101 , thanks for your response.

Please check the A/B relation fields setting in Log model, on line 5 and line 6 in scheme block above.

I want to get the relation logs like below:
```
// Get Logs which type='a' and link_id = A.id
Prisma.A.findMany({
  include: {Logs: true}},
})

// OR Get Logs which type = 'b' and link_id=B.id 
Prisma.B.findMany({
  include: {Logs: true}},
})
```

If it is possible, so I do not have to create ALogs/BLogs/CLogs model for A/B, I just need a Log model for A/B/C with Log.type.

**Reply by nurul3101** [reply, marked as answer]
You would need to model your relations like these:
```prisma
model Log {
  id        Int    @id @default(autoincrement())
  type      String @db.VarChar(16)
  a_link_id Int?
  b_link_id Int?
  A         A?     @relation("aLog", fields: [a_link_id], references: [id]) // here
  B         B?     @relation("bLog", fields: [b_link_id], references: [id])
}

model A {
  id   Int   @id @default(autoincrement())
  Logs Log[] @relation("aLog")
}

model B {
  id   Int   @id @default(autoincrement())
  Logs Log[] @relation("bLog")
}
```
Two relations cannot have the same fields, you would need to define two fields `a_link_id` and `b_link_id` and then you can filter on them.

**Reply by Riant** [reply]
Yes, I knew that, but it looks ugly.

**Reply by nurul3101** [reply]
At the moment, this is the only possible solution. I'll share the feedback with the team so that we can work on improving the experience.

**Reply by Riant** [reply]
Thanks.
